core/src/main/scala/kafka/tools/StreamsResetter.java

```diff
@@ -313,10 +316,12 @@ private int maybeResetInputAndSeekToEndIntermediateTopicOffsets(final Map consum
         config.setProperty(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "false");
 
         try (final KafkaConsumer<byte[], byte[]> client = new KafkaConsumer<>(config, new ByteArrayDeserializer(), new ByteArrayDeserializer())) {
-            client.subscribe(topicsToSubscribe);
-            client.poll(1);
+            Collection<TopicPartition> partitions = topicsToSubscribe.stream().map(client::partitionsFor)
```

Comment:
@hachikuji were you thinking of something like this?

Reply:
LGTM.

Reply:
Java 8 makes this much nicer than I was expecting. 😄